src/uct/tcp/tcp_ep.c

```diff
@@ -551,6 +580,18 @@ static void uct_tcp_ep_handle_disconnected(uct_tcp_ep_t *ep,
 
         uct_tcp_ep_mod_events(ep, 0, ep->events);
         ucs_close_fd(&ep->fd);
+
+        if (ep->ctx_caps & UCS_BIT(UCT_TCP_EP_CTX_TYPE_ZCOPY_TX)) {
+            /* There is ongoing AM/PUT Zcopy operation, need to notify
+             * the user about the error */
+            ctx = (uct_tcp_ep_zcopy_tx_t*)ep->tx.buf;
+            uct_tcp_ep_comp_zcopy(ep, ctx->comp, status);
+        }
+
+        uct_tcp_ep_comp_tx(ep, ep->tx.length - ep->tx.offset);
+        if (ep->conn_state == UCT_TCP_EP_CONN_STATE_CONNECTED) {
```

Comment:
what will be if failure occurs while you are waiting for conn_req ack?

Reply:
removed this check